src/Language/Haskell/Liquid/GHC/Plugin.hs

```diff
@@ -334,29 +343,31 @@ liquidHaskellCheck pipelineData modSummary tcGblEnv = do
     file :: FilePath
     file = LH.modSummaryHsFile modSummary
 
-    continue :: TcM (Either LiquidCheckException TcGblEnv)
+    continue :: TcM (Either LiquidCheckException (TcGblEnv, [Filter]))
     continue = pure $ Left KillPluginWithSuccess
 
-    reportErrs :: (Show e, F.PPrint e) => Config -> [TError e] -> TcM (Either LiquidCheckException TcGblEnv)
+    reportErrs :: (Show e, F.PPrint e) => Config -> [TError e] -> TcM (Either LiquidCheckException (TcGblEnv, [Filter]))
     reportErrs cfg = LH.filterReportErrors file GHC.failM continue (getFilters cfg) Full
 
-checkLiquidHaskellContext :: LiquidHaskellContext -> TcM (Either LiquidCheckException LiquidLib)
+checkLiquidHaskellContext :: LiquidHaskellContext -> TcM (Either LiquidCheckException (LiquidLib, [Filter]))
 checkLiquidHaskellContext lhContext = do
-  ProcessModuleResult{..} <- processModule lhContext
-
-  -- Call into the existing Liquid interface
-  out <- liftIO $ LH.checkTargetInfo pmrTargetInfo
-
-  let bareSpec = lhInputSpec lhContext
-      file = LH.modSummaryHsFile $ lhModuleSummary lhContext
-
-  withPragmas (lhGlobalCfg lhContext) file (Ms.pragmas $ review bareSpecIso bareSpec) $ \moduleCfg ->  do
-    let filters = getFilters moduleCfg
-    -- Report the outcome of the checking
-    wasUnsafeButContinued <- LH.reportResult (errorLogger file filters) moduleCfg [giTarget (giSrc pmrTargetInfo)] out
-    if wasUnsafeButContinued
-      then pure $ Left KillPluginWithSuccess
-      else pure $ Right pmrClientLib
+  pmr <- processModule lhContext
+  case pmr of
+    Left e -> pure $ Left e
+    Right ProcessModuleResult{..} -> do
+      -- Call into the existing Liquid interface
+      out <- liftIO $ LH.checkTargetInfo pmrTargetInfo
+
+      let bareSpec = lhInputSpec lhContext
+          file = LH.modSummaryHsFile $ lhModuleSummary lhContext
+
+      withPragmas (lhGlobalCfg lhContext) file (Ms.pragmas $ review bareSpecIso bareSpec) $ \moduleCfg ->  do
+        let filters = getFilters moduleCfg
+        -- Report the outcome of the checking
+        wasUnsafeButContinued <- LH.reportResult (errorLogger file filters) moduleCfg [giTarget (giSrc pmrTargetInfo)] out
+        pure $ if wasUnsafeButContinued
+          then Left KillPluginWithSuccess
+          else Right (pmrClientLib, filters)
```

Comment:
Perhaps the unmatched filters should be preserved when killing the plugin with success.